selfdrive/car/tests/test_models.py

```diff
@@ -201,6 +201,42 @@ def test_panda_safety_rx_valid(self):
           self.assertTrue(self.safety.addr_checks_valid())
     self.assertFalse(len(failed_addrs), f"panda safety RX check failed: {failed_addrs}")
 
+  def test_panda_safety_tx_cases(self, data=None):
+    """Asserts we can tx common messages"""
+    if self.CP.notCar:
+      self.skipTest("Skipping test for notCar")
+
+    def test_car_controller(car_control, _now_nanos):
+      msgs_sent = 0
+      for _ in range(300):  # make sure we hit the slowest messages
+        self.CI.update(car_control, [])
+        _, sendcan = self.CI.apply(car_control, _now_nanos)
+        _now_nanos += 10e6
+        msgs_sent += len(sendcan)
+        for addr, _, dat, bus in sendcan:
+          to_send = libpanda_py.make_CANPacket(addr, bus % 4, dat)
+          self.assertTrue(self.safety.safety_tx_hook(to_send), (addr, dat, bus))
+
+      # Make sure we attempted to send messages
+      self.assertGreater(msgs_sent, 50)
+      return _now_nanos
```

Comment:
no need to return or pass this in, just use a nonlocal

Reply:
or even better, make a new self.CI so that it's scope is just within the function